pkg/api/types.go

```diff
@@ -1299,6 +1299,11 @@ func (m *MasterProfile) IsVHDDistro() bool {
 	return m.Distro == AKSUbuntu1604 || m.Distro == AKSUbuntu1804
 }
 
+// IsAuditDEnabled returns true if the master profile is configured for auditd
+func (m *MasterProfile) IsAuditDEnabled() bool {
+	return m.AuditDEnabled != nil && to.Bool(m.AuditDEnabled)
```

Comment:
A `nil` check isn't necessary since [`to.Bool`](https://github.com/Azure/go-autorest/blob/master/autorest/to/convert.go#L71) returns `false` in that case.